### PR TITLE
Convert and copy all translations at build time

### LINKED ISSUE #10: Pull in translations from Transifex
This library should pull translations directly from Transifex and convert them into plist format to complement the existing English plist. This should be done as a build step to keep developers from having to access the Transifex API every time they run `carthage bootstrap`.

/cc @frederoni @freenerd

### PR DESCRIPTION
**Pull request by 1ec5**
This PR improves the process of producing the project’s plists, preventing us from having to push PRs for each change to translations in the main osrm-text-instructions repository. Also, every language included by osrm-text-instructions is also included in this port.

A script that runs as part of the build (for Carthage users) or as a preflight step (for CocoaPods users) updates the submodule, converts the translation JSON files into plist format, then copies the plists into the built product bundle. The checked-in instructions plists have been deleted, since they can be regenerated on each build.

Fixes #10.

/cc @bsudekum @frederoni

### CODE REVIEW

**Comment by 1ec5** [comment]
Before merging this PR, I’d like to merge #13, which fixes the CocoaPods podspec.

**Comment by 1ec5** [comment]
This PR also pulls in Project-OSRM/osrm-text-instructions@b4c5187b6635a83bcd0b417fb507366ef8aa5fd2. All in all, the following languages have been added:

* Dutch
* French
* German
* Russian
* Swedish
* Vietnamese

**Comment by 1ec5** [comment]
The CocoaPods podspec linter is erroring out because the `resources` path no longer matches any checked-in files:

```
    - ERROR | file patterns: The `resources` pattern did not match any file.
```

Some of the changes to the conversion script in 7f2eadc70f85f951647222b756f49bb0436094d9 will need to be conditionalized: in Xcode (for Carthage), the files would be written into the build product; when run standalone (for CocoaPods), the files would be written to an easily accessed location, such as under OSRMTextInstructions/*.lproj/.

**Comment by 1ec5** [comment]
ef3fdec9e912d41131fb4231985edf1f35efbb54 fixes the CocoaPods installation error.

File: json2plist.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+for file in ./*; do
+    if [ "$file" = "./en.json" ]; then
+      LANGUAGE="Base"
+    else
+      # skip not-supported languages
+      LANGUAGE=$(basename $file)
+      LANGUAGE=${LANGUAGE%.json}
+    fi
+
+    if [ "$LANGUAGE" != "skip" ]; then
```
Comment (freenerd) [inline json2plist.sh]: when will language ever be `skip`?

**Comment by freenerd** [comment]
I'm good with this approach. The only downside I see is that the git submodule checked into .git as part of osrm-text-instructions.swift will over time very much fall behind. This may be confusing for users, given that they see a submodule that is not actually used. Instead, it might make sense to the submodule only in `json2plist.sh` and remove it after again? This way the working git directory stays clean.

**Comment by freenerd** [comment]
On second thought I think this PR is a bad idea.

When we do structural changes on the language files in `osrm-text-instructions` we will release a new version there. `osrm-text-instructions.swift` has no concept of versions but pulls in `master`. This means that it would pull in these new files and would break (or even worse: have undetected side-effects). This happens without any change in `osrm-text-instructions.swift` and is therefore a surprising side-effect. Instead, there should be a PR in `osrm-text-instructions.swift` with the new version (and a passing test suite).

I understand that the premise of this PR is to speed-up the process of changes in Transifex being available in `osrm-text-instructions.swift`, but the risk of the approach here seems greater than that benefit.

File: json2plist.sh (hunk 1 above)
Comment (1ec5) [inline json2plist.sh]: Good catch. That can be removed.

**Comment by 1ec5** [comment]
The risk associated with structural changes upstream should be no different than it is now on master. Note that `git submodule update` doesn't change the commit that the submodule points to; it only ensures that the local clone points to the correct commit. (To float on master, the script would have to `cd` into the submodule directory and `git checkout master`.)

**Comment by 1ec5** [comment]
@freenerd, as a case in point, if you run the script in this PR, you’ll wind up with Project-OSRM/osrm-text-instructions@b4c5187b6635a83bcd0b417fb507366ef8aa5fd2, not the latest master (currently Project-OSRM/osrm-text-instructions@6a57291d2cf530f067637ea720c3f10c220b25e6).

**Comment by freenerd** [comment]
@1ec5 thanks for the clarification, this makes sense.

**Comment by 1ec5** [comment]
Yay! By the way, the build failures are because .travis.yml won’t exist until #17 lands, but Travis (unlike Bitrise) insists that the file has to exist on every branch until it does. Only a temporary inconvenience.

**Pull request merged**